libevmasm/AssemblyItem.h

```diff
@@ -170,8 +172,24 @@ class AssemblyItem
 	/// @returns true if the assembly item can be used in a functional context.
 	bool canBeFunctional() const;
 
-	void setLocation(langutil::SourceLocation const& _location) { m_location = _location; }
-	langutil::SourceLocation const& location() const { return m_location; }
+	void setLocation(langutil::SourceLocation const& _location)
+	{
+		solAssert(m_debugData);
+		m_debugData = langutil::DebugData::create(
+			_location,
+			m_debugData->originLocation,
+			m_debugData->astID
+		);
+	}
+
+	langutil::SourceLocation const& location() const
+	{
+		solAssert(m_debugData);
+		return m_debugData->nativeLocation;
+	}
+
+	void setDebugData(langutil::DebugData::ConstPtr const& _debugData) { m_debugData = _debugData; }
```

Comment:
```suggestion
	void setDebugData(langutil::DebugData::ConstPtr const& _debugData)
	{
		solAssert(_debugData);
		m_debugData = _debugData;
	}
```
if we want to guarantee that there's always a valid one there, we also need to do it here.

Reply:
If you want, while you're at it: passing the pointer by value and moving it is, I think, a bit more idiomatic for shared pointers... but also doesn't matter too much.

Reply:
You're right! I change that only where it is possible to use `std::move` - e.g. if there is exactly one other place where this pointer can be moved. I let it be a constant reference, when the pointer is used to set multiple things.